src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T element) {
+            this.element = element;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (size == 0) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " is out of list bound");
+        }
+
+        if (head == null) {
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
```

Comment:
```suggestion
        if (index == size) {
```
if index == 0 and size == 0 -> index == size
so add(value) will be called